lib/liquid/context.rb

```diff
@@ -166,24 +146,9 @@ def find_variable(key, raise_on_not_found: true)
       index = @scopes.find_index { |s| s.key?(key) }
       scope = @scopes[index] if index
 
-      variable = nil
-
-      if scope.nil?
-        @environments.each do |e|
-          variable = lookup_and_evaluate(e, key, raise_on_not_found: raise_on_not_found)
-          # When lookup returned a value OR there is no value but the lookup also did not raise
-          # then it is the value we are looking for.
-          if !variable.nil? || @strict_variables && raise_on_not_found
-            scope = e
-            break
-          end
-        end
-      end
-
-      scope ||= @environments.last || @scopes.last
-      variable ||= lookup_and_evaluate(scope, key, raise_on_not_found: raise_on_not_found)
+      scope ||= @environments.find { |e| !e[key].nil? || @strict_variables && raise_on_not_found } || {}
```

Comment:
You might want to use `find_index` here instead of `find` (same as above at the top of this method)?